roles/destroy_vms/tasks/main.yml

```diff
@@ -1,33 +1,40 @@
-- name: Remove network
+- name: Create inventory host to vm name mapper
+  set_fact:
+    cluster_vm_names: "{{ (cluster_vm_names | default([])) + [item.startswith(cluster_name) | ternary(item, (cluster_name + '_' + item))] }}"
+  loop: "{{ groups['masters'] + groups['workers'] | default([]) }}"
+  when: hostvars[item]['vendor'] | lower == 'kvm'
+
+- debug: # noqa unnamed-task
+    var: vm_to_inventory_mapper
+    verbosity: 1
+
+- name: Destroy vms
+  import_tasks: destroy_networks.yml
   become: true
-  ignore_errors: yes
+
+- name: Get VMs to Remove
   block:
-    - name: destroy network
-      community.libvirt.virt_net:
-        name: "{{ network_name }}"
-        command: destroy
+    - name: List vms
+      community.libvirt.virt:
+        command: list_vms
+      register: vms_list
+      become: true
 
-    - name: undefine network
-      community.libvirt.virt_net:
-        name: "{{ network_name }}"
-        command: undefine
+    - name: VM list
+      debug:
+        var: vms_list.list_vms
+        verbosity: 1
 
-- name: Delete existing bridges (if any)
-  community.general.nmcli:
-    conn_name: "{{ item }}"
-    type: bridge
-    state: absent
-  loop:
-    - "{{ bridge_name }}"
-    - "{{ vm_bridge_interface }}_{{ bridge_name }}"
-  ignore_errors: yes
-  become: true
-  when: (SETUP_VM_BRIDGE | default(true)) | bool == true
+    - name: Get filtered vm list
+      set_fact:
+        vms_to_remove: "{{ (vms_to_remove | default([])) + [item] }}"
+      loop: "{{ vms_list.list_vms }}"
+      when: item in cluster_vm_names
```

Comment:
I'd be cool if we could add this information to the deployment plan, as to make the user aware what VMs will be removed once Crucible is run. This might be out of scope for this PR, though.